packages/@sanity/core/src/actions/deploy/undeployAction.js

```diff
@@ -0,0 +1,51 @@
+export default async (args, context) => {
+  const {apiClient, chalk, output, prompt} = context
+
+  const client = apiClient({
+    requireUser: true,
+    requireProject: true
+  })
+
+  // Check that the project has a studio hostname
+  let spinner = output.spinner('Checking project info').start()
+  const project = await client.projects.getById(client.config().projectId)
+  const studioHostname = project && (project.studioHostname || project.studioHost)
+  spinner.succeed()
+
+  if (!studioHostname) {
+    output.print('Your project has not been assigned a studio hostname.')
+    output.print('Nothing to undeploy.')
+    return
+  }
+
+  // Double-check
+  output.print('')
+
+  const shouldUndeploy = await prompt.single({
+    type: 'confirm',
+    default: false,
+    message: `
+https://${chalk.yellow(studioHostname)}.sanity.studio will be available for anyone to claim.
```

Comment:
Suggestion: "This will undeploy https://${chalk.yellow(studioHostname)}.sanity.studio and make it unavailable for your users. The hostname will be available for anyone to claim."

Reply:
✅